instrumentation/message-handler/library/src/main/java/io/opentelemetry/instrumentation/messagehandler/SqsBatchMessageHandler.java

```diff
@@ -0,0 +1,117 @@
+import io.opentelemetry.context.propagation.TextMapGetter;
+import io.opentelemetry.instrumentation.api.instrumenter.AttributesExtractor;
+import io.opentelemetry.instrumentation.api.instrumenter.Instrumenter;
+import io.opentelemetry.instrumentation.api.instrumenter.InstrumenterBuilder;
+import io.opentelemetry.instrumentation.api.instrumenter.SpanLinksExtractor;
+import io.opentelemetry.instrumentation.api.instrumenter.SpanNameExtractor;
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Locale;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+public abstract class SqsBatchMessageHandler extends BatchMessageHandler<SQSEvent.SQSMessage> {
+  private static final String AWS_TRACE_HEADER_SQS_ATTRIBUTE_KEY = "AWSTraceHeader";
+
+  // lower-case map getter used for extraction
+  static final String AWS_TRACE_HEADER_PROPAGATOR_KEY = "x-amzn-trace-id";
+
+  public SqsBatchMessageHandler(OpenTelemetry openTelemetry) {
+    super(openTelemetry);
+  }
+
+  public SqsBatchMessageHandler(OpenTelemetry openTelemetry, String messageOperation) {
+    super(openTelemetry, messageOperation);
+  }
+
+  public SqsBatchMessageHandler(
+      OpenTelemetry openTelemetry,
+      String messageOperation,
+      SpanNameExtractor<Collection<SQSEvent.SQSMessage>> spanNameExtractor) {
+    super(openTelemetry, messageOperation, spanNameExtractor);
+  }
+
+  @Override
+  protected void setup() {
+    InstrumenterBuilder<Collection<SQSEvent.SQSMessage>, Void> builder =
+        Instrumenter.builder(openTelemetry, "io.opentelemetry.message.handler", spanNameExtractor);
+
+    builder.setInstrumentationVersion("1.0");
+    builder.addAttributesExtractor(getGenericAttributesExtractor());
+    builder.addAttributesExtractor(getAttributesExtractor());
+    builder.addSpanLinksExtractor(getSpanLinksExtractor());
+
+    messageInstrumenter = builder.buildInstrumenter();
+  }
+
+  protected AttributesExtractor<Collection<SQSEvent.SQSMessage>, Void> getAttributesExtractor() {
+    return new AttributesExtractor<Collection<SQSEvent.SQSMessage>, Void>() {
+
+      @Override
+      public void onStart(
+          AttributesBuilder attributes,
+          Context parentContext,
+          Collection<SQSEvent.SQSMessage> messages) {
+        attributes.put(SemanticAttributes.MESSAGING_SYSTEM, "AmazonSQS");
+      }
+
+      @Override
+      public void onEnd(
+          AttributesBuilder attributes,
+          Context context,
+          Collection<SQSEvent.SQSMessage> messages,
+          @Nullable Void unused,
+          @Nullable Throwable error) {}
+    };
+  }
+
+  protected SpanLinksExtractor<Collection<SQSEvent.SQSMessage>> getSpanLinksExtractor() {
+    return (spanLinks, parentContext, sqsMessages) -> {
+      for (SQSEvent.SQSMessage message : sqsMessages) {
+        String parentHeader = message.getAttributes().get(AWS_TRACE_HEADER_SQS_ATTRIBUTE_KEY);
```

Comment:
There should be an option to use the configured propagator as well (see #8405). On receive, it should maybe even be enabled by default.

Reply:
@Oberon00 
I am not sure what the ask is here. There is a lot going on in that PR.

Are you asking for the option to change the propagator say to one that is W3C compatible by passing one into MessageHandler? Can the customer not change the global one programmatically?

Reply:
Oh I see: You are calling the global propagator, that can indeed be changed, but you are calling it hardcoded with only the AWS_TRACE_HEADER_SQS_ATTRIBUTE_KEY. This is wrong in any case I think. You should not hardcode AWS_TRACE_HEADER_SQS_ATTRIBUTE_KEY but instead pass the propagator the full message attributes as carrier with a getter that is able to fetch any string attributes the propagator requests.

Reply:
I see. Updated the PR.